docs/docker.md

```diff
@@ -0,0 +1,101 @@
+
+```
+$ docker run -it -e "GHE_HOSTNAME=hostname" \
+-e "GHE_DATA_DIR=/data" \
+-e "GHE_EXTRA_SSH_OPTS=-i /ghe-ssh/id_rsa -o UserKnownHostsFile=/ghe-ssh/known_hosts \
+-e "GHE_NUM_SNAPSHOTS=15" \
+-v ghe-backup-data:/data \
+-v "$HOME/.ssh/known_hosts:/ghe-ssh/known_hosts" \
+-v "$HOME/.ssh/id_rsa:/ghe-ssh/id_rsa" \
+--rm \
+github/backup-utils ghe-backup
+```
+
+It is also possible to specify a `-e GHE_BACKUP_CONFIG` flag and volume mount in a local `backup.config` file rather than specify the variables individually at run time:
+
+```
+$ docker run -it  -e "GHE_BACKUP_CONFIG=/mnt/backup.config" \
+-v ghe-backup-data:/data \
+-v "$HOME/.ssh/known_hosts:/ghe-ssh/known_hosts" \
+-v "$HOME/.ssh/id_rsa:/ghe-ssh/id_rsa" \
+-v $HOME/backup-utils/backup.config:/mnt/backup.config \
+--rm \
+github/backup-utils ghe-backup
+```
+
+#### SSH Keys
+
+A SSH private key that has been added to the GitHub Enterprise [Management Console for administrative SSH access](https://help.github.com/enterprise/admin/guides/installation/administrative-shell-ssh-access/) needs to be mounted into the container from the host system. It is also recommended to mount a SSH `.ssh/known_hosts` file into the container.
+
+```
+$ docker run -it -e "GHE_HOSTNAME=hostname" \
+-e "GHE_DATA_DIR=/data" \
+-e "GHE_EXTRA_SSH_OPTS=-i /ghe-ssh/id_rsa -o UserKnownHostsFile=/ghe-ssh/known_hosts \
+-v ghe-backup-data:/data \
+-v "$HOME/.ssh/known_hosts:/ghe-ssh/known_hosts" \
+-v "$HOME/.ssh/id_rsa:/ghe-ssh/id_rsa" \
+--rm \
+github/backup-utils ghe-backup
+```
+
+#### Managing backup data
+
+Data persistence is achieved by using [Docker volumes](https://docs.docker.com/engine/admin/volumes/volumes/), which are managed with [`docker volume` commands](https://docs.docker.com/engine/reference/commandline/volume/). Prior to running the container for the first time, a volume can be created if you need to specify additional options. The named volume will be automatically created at runtime if it does not exist:
+
+```
+docker volume create ghe-backup-data
+```
+
+The named Docker volume can be mounted and accessed from other containers, using any image you like:
+
+```
+# Accessing backups using the backup-utils image:
+
+$ docker run -it -v ghe-backup-data:/data --rm github/backup-utils ls -l /data/
+total 8
+drwxr-xr-x 11 root root 4096 Oct 24 19:46 20171024T194650
+drwxr-xr-x 11 root root 4096 Oct 24 19:49 20171024T194921
+lrwxrwxrwx  1 root root   15 Oct 24 19:49 current -> 20171024T194921
+
+# Accessing backups using the busybox library image:
+
+$ docker run --rm -v ghe-backup-data:/data busybox ls -l /data
+total 8
+drwxr-xr-x   11 root     root          4096 Oct 24 19:46 20171024T194650
+drwxr-xr-x   11 root     root          4096 Oct 24 19:49 20171024T194921
+lrwxrwxrwx    1 root     root            15 Oct 24 19:49 current -> 20171024T194921
+```
+
+The volume's filesystem must support hard links. This works fine on Docker for Mac with the default local driver in my testing so far.
```

Comment:
I think we should remove your experience comment here. We could go further to state something like "tested on X, Y & Z" but I'd rather hold off on that until we've got an more precise list of supported operating systems.